packages/compass-connections/src/components/legacy-connections.tsx

```diff
@@ -88,89 +115,69 @@ function Connections({
   const { log, mongoLogId } = useLogger('COMPASS-CONNECTIONS');
 
   const {
-    state,
-    cancelConnectionAttempt,
+    state: { editingConnectionInfo, connectionErrors, oidcDeviceAuthState },
     connect,
+    disconnect,
     createNewConnection,
-    legacyDuplicateConnection: duplicateConnection,
-    setActiveConnectionById,
-    removeAllRecentsConnections,
+    editConnection,
+    duplicateConnection,
+    removeAllRecentConnections,
     removeConnection,
-    saveConnection,
-    favoriteConnections,
-    recentConnections,
+    saveEditedConnection,
   } = useConnections();
 
-  const {
-    connectingConnectionId,
-    activeConnectionId,
-    activeConnectionInfo,
-    connectionErrorMessage,
-    connectingStatusText,
-    oidcDeviceAuthVerificationUrl,
-    oidcDeviceAuthUserCode,
-  } = state;
+  const { favoriteConnections, nonFavoriteConnections: recentConnections } =
+    useConnectionRepository();
 
   const darkMode = useDarkMode();
-
-  const protectConnectionStrings = usePreference('protectConnectionStrings');
-  const forceConnectionOptions = usePreference('forceConnectionOptions');
-  const showKerberosPasswordField = usePreference('showKerberosPasswordField');
-  const showOIDCDeviceAuthFlow = usePreference('showOIDCDeviceAuthFlow');
-  const enableOidc = usePreference('enableOidc');
-  const enableDebugUseCsfleSchemaMap = usePreference(
-    'enableDebugUseCsfleSchemaMap'
-  );
-  const protectConnectionStringsForNewConnections = usePreference(
-    'protectConnectionStringsForNewConnections'
-  );
+  const connectionFormPreferences = useConnectionFormPreferences();
   const isMultiConnectionEnabled = usePreference(
     'enableNewMultipleConnectionSystem'
   );
 
-  const preferences = useMemo(
-    () => ({
-      protectConnectionStrings,
-      forceConnectionOptions,
-      showKerberosPasswordField,
-      showOIDCDeviceAuthFlow,
-      enableOidc,
-      enableDebugUseCsfleSchemaMap,
-      protectConnectionStringsForNewConnections,
-    }),
-    [
-      protectConnectionStrings,
-      forceConnectionOptions,
-      showKerberosPasswordField,
-      showOIDCDeviceAuthFlow,
-      enableOidc,
-      enableDebugUseCsfleSchemaMap,
-      protectConnectionStringsForNewConnections,
-    ]
+  const activeConnectionInfo = useActiveConnectionInfo(editingConnectionInfo);
+  const activeConnectionStatus = useConnectionInfoStatus(
+    activeConnectionInfo.id
   );
 
   const onConnectClick = (connectionInfo: ConnectionInfo) => {
-    void connect({ ...cloneDeep(connectionInfo) }, true).catch(() => {
```

Comment:
Honestly not sure why this was shallow and deep cloning, if someone knows what's the reason for this, please tell